cpp/src/rolling/grouped_rolling.cu

```diff
@@ -699,20 +707,13 @@ std::unique_ptr<column> time_range_window_DESC(
            1;
   };
 
+  auto following_column = expand_to_column(following_calculator, input.size(), mr);
+
   if (aggr->kind == aggregation::CUDA || aggr->kind == aggregation::PTX) {
     CUDF_FAIL("Time ranged rolling window does NOT (yet) support UDF.");
   } else {
-    return cudf::detail::rolling_window(
-      input,
-      empty_like(input)->view(),
-      thrust::make_transform_iterator(thrust::make_counting_iterator<size_type>(0),
-                                      preceding_calculator),
-      thrust::make_transform_iterator(thrust::make_counting_iterator<size_type>(0),
-                                      following_calculator),
-      min_periods,
-      aggr,
-      rmm::cuda_stream_default,
-      mr);
+    return cudf::rolling_window(
+      input, preceding_column->view(), following_column->view(), min_periods, aggr, mr);
```

Comment:
Not directly related to this PR (just the fact that we call the public API here), but I'm a bit surprised at how deep the call stack is before a `detail::` function is called. Should some of these functions be moved to `detail` namespace?

Reply:
These are all functions in the unnamed namespace under `cudf`. Would you recommend that these be moved to the unnamed namespace under `cudf::detail`?

Reply:
I'm not sure what's the right way to organize them. The reason I noticed is that we aren't adding a stream parameter until we're deep in the call stack. My understanding is that public APIs have a detail:: counterpart that has a stream parameter. It's likely that I'm misunderstanding/overgeneralizing something here, but wanted to bring it up just in case.

Reply:
>  My understanding is that public APIs have a detail:: counterpart that has a stream parameter

That is correct:
1. Helper function:`cudf::detail::rolling_window()` accepts a stream. 
2. Public functions: `cudf::rolling_window()` and `cudf::*grouped_rolling_window` do not accept a stream, and delegate to `cudf::detail::rolling_window()` with `cudf_default_stream`.

There are private helper functions in `grouped_rolling.cu` that also do not accept streams. I have put those in the unnamed namespace, since they aren't currently meant for reuse. That might change later, though, in ways that I haven't yet anticipated. I'd prefer they remain with internal linkage until they're required elsewhere, but I'd be happy to move them into `cudf::detail::<unnamed>` if that is preferable.

Please pardon me if I haven't grokked your advice.

Reply:
I guess my expectation was that
```
std::unique_ptr<column> grouped_time_range_rolling_window(table_view const& group_keys,
                                                          column_view const& timestamp_column,
                                                          cudf::order const& timestamp_order,
                                                          column_view const& input,
                                                          window_bounds preceding_window_in_days,
                                                          window_bounds following_window_in_days,
                                                          size_type min_periods,
                                                          std::unique_ptr<aggregation> const& aggr,
                                                          rmm::mr::device_memory_resource* mr)
```
and everything it calls would accept the stream parameter. 
As you said, this can be changed later. Which is fine, I "request changes"'d only because of the missing newline :D

Reply:
> `grouped_time_range_rolling_window(...)` and everything it calls would accept the stream parameter.

I quite agree. I'm not sure why the `rolling_window()` public API functions don't accept a `stream` parameter. They seem to adopt `cudf_default_stream` halfway down. I followed that convention when I came on the scene with `*grouped_rolling_window()`. I suppose it'd be a breaking change to add them now. :/

> I "request changes"'d only because of the missing newline :D

Argh, that was an oversight on my part. :] I'll correct that shortly.

Reply:
Public functions should NOT take a stream. But they should immediately call an equivalent detail APIs that DO take a stream, and pass `rmm::cuda_stream_default`.

Reply:
Thanks, @harrism. @vuule alerted me to that section of the `TRANSITIONGUIDE.md` as well. There are a number of public functions in `rolling.cu` now that do not follow that convention. Let me try sort those out.